flake8_trio/runner.py

```diff
@@ -21,44 +21,48 @@
 )
 
 if TYPE_CHECKING:
-    from argparse import Namespace
     from collections.abc import Iterable
 
     from libcst import Module
 
-    from .base import Error
+    from .base import Error, Options
     from .visitors.flake8triovisitor import Flake8TrioVisitor, Flake8TrioVisitor_cst
 
 
 @dataclass
 class SharedState:
-    options: Namespace
+    options: Options
     problems: list[Error] = field(default_factory=list)
     library: tuple[str, ...] = ()
     typed_calls: dict[str, str] = field(default_factory=dict)
     variables: dict[str, str] = field(default_factory=dict)
 
 
-class Flake8TrioRunner(ast.NodeVisitor):
-    def __init__(self, options: Namespace):
+class CommonRunner(ABC):  # noqa: B024 # no abstract methods
+    def __init__(self, options: Options):
         super().__init__()
         self.state = SharedState(options)
 
+    def selected(self, error_codes: dict[str, str]) -> bool:
+        for code in error_codes:
+            for enabled in self.state.options.enable, self.state.options.autofix:
+                if code in enabled:
+                    return True
+        return False
```

Comment:
I find it easier to think about this in terms of set operations:

```suggestion
        enabled_codes = set(self.state.options.enable) | set(self.state.options.autofix)
        return bool(enabled_codes & set(error_codes))
```